src/lib/OS/SyscallHandler.cc

```diff
@@ -488,20 +494,60 @@ void SyscallHandler::handleSyscall() {
         if (bitmask != 1) {
           return concludeSyscall({}, true);
         }
-        uint64_t retval = (pid == 0) ? 1 : 0;
-        stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {retval}};
-        stateChange.memoryAddresses.push_back({mask, 1});
+        uint64_t retval = static_cast<uint64_t>(bitmask);
+        stateChange = {
+            ChangeType::REPLACEMENT, {currentInfo_.ret}, {sizeof(retval)}};
+        stateChange.memoryAddresses.push_back({mask, 8});
         stateChange.memoryAddressValues.push_back(bitmask);
       } else {
         stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {-1ll}};
       }
       break;
     }
+    case 124: {  // sched_yield
+      // Non args passed in
+      // Have core go to idle after syscall, forcing the current Process to be
+      // de-scheduled
+      return concludeSyscall({}, false, true);
+    }
     case 131: {  // tgkill
-      // TODO: Functionality temporarily omitted since simeng only has a
-      // single thread at the moment
-      stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
-      break;
+      int tgid = currentInfo_.registerArguments[0].get<int>();
+      int tid = currentInfo_.registerArguments[1].get<int>();
+      int signal = currentInfo_.registerArguments[2].get<int>();
+      int64_t retVal = 0;
+      bool idleOnComplete = false;
+
+      // Only support SIGABORT or no signal
+      if (signal != SIGABRT || signal != 0) {
+        retVal = -EINVAL;
+      } else {
+        auto proc = OS_->getProcess(tid);
+        uint64_t procTgid = proc->getTGID();
+        if (tgid == -1) {
+          // Terminate all processes in thread group
+          OS_->terminateThreadGroup(procTgid);
+          std::cout << "[SimEng:SyscallHandler] Received tgkill syscall on "
+                       "Thread Group "
+                    << procTgid << ". Terminating with signal " << signal
+                    << std::endl;
+          idleOnComplete = true;
+        } else {
+          if (proc->getTGID() == tgid) {
+            if (proc->status_ == procStatus::executing) {
+              idleOnComplete = true;
+            }
+            OS_->terminateThread(tid);
+            std::cout
+                << "[SimEng:SyscallHandler] Received tgkill syscall on Thread "
+                << tid << " in Thread Group " << tgid
+                << ". Terminating with signal " << signal << std::endl;
+          } else {
+            retVal = -ESRCH;
```

Comment:
Is this value consistent across macOS and Linux, If it is please resolve the comment. Thanks.

Reply:
MacOS and Linux values match (both 3)

Reply:
Do you think these should be moved to Constants.hh for consistency?

Reply:
I don't think so. May cause confusion as the libary defined values will work